### ISSUE TITLE
Support v3

### ISSUE THREAD

**Issue by lavellotron** [sarneeh/reaptcha#117]
The repo description mentions v2 but isn't this compatible with v3 reCaptcha actually?

**Comment by sarneeh** [comment]
@lavellotron Unfortunately not. 

I did a little bit of research to answer your question and found that there are few differences in v2 and v3:
* v3 has a different render API (it needs a `render=SITE_KEY` query param in the external script load)
* v3 accepts a `action` parameter in the `execute` function
* v3 `execute` function returns a promise
* v3 does not render anything into the DOM (there's no need to render any element by the component)

I'd like to add support for v3 in `reaptcha` but I'm out of time right now. I'll keep this issue and notes above for easier implementation later on. Of course, if anyone sees this comment and would like to contribute - feel free to do so 😄

**Comment by MrComic** [comment]
any chance to add the action name feature? that will help us to analyze the verification requests better on v3.

**Comment by sarneeh** [comment]
@MrComic When I'll find time to implement this, I'll try to support every feature from v3.

**Comment by jasonwilczak** [comment]
Hi there, curious if v3 support will be coming to this package?

**Comment by sarneeh** [comment]
> Hi there, curious if v3 support will be coming to this package?

Yes, but I'm still out of time 🙈  I want to implement this and rewrite everything to TypeScript so I need to find a bigger time slot to do this. Certainly not in the nearest months, but it will be there this year for sure.

**Comment by pyon123** [comment]
Hi @sarneeh 
Thanks for writing this cool repo. Btw, would you like to let me know when this repo can support V3? it seems like the react-google-recaptcha doesn't provide a good api.

**Comment by sarneeh** [comment]
@pyon123 I'll be working on it in the next few weeks

**Comment by pyon123** [comment]
thank you @sarneeh

**Comment by jakerobb** [comment]
Hi, is there an update on this? It's been a few weeks. ;)

**Comment by sarneeh** [comment]
@jakerobb I don't have anything yet. No time or mood to work on it to be honest 😄
